benchmark/matmul.cpp

```diff
@@ -297,6 +343,23 @@ static void NvFuserScheduler_Matmul_8warp4stage(
 
 // ----------------------------- Benchmark Instantiation-------
 
+// For 4warp splitk experiments, we use a tile size of (128, 128). To avoid
+// wave quantization, we look at sizes that are integer multiples of the block
+// size. Below you will find all the factors of 108, which is the number of SMs
+// on an A100. Note that 8warp uses tile size (256, 128) in which case SplitKMs
+// should be changed to 256.
+#define SplitKMs \
+  { 128 }
+#define SplitKNs                                                              \
+  {                                                                           \
+    1 * 128, 2 * 128, 3 * 128, 4 * 128, 6 * 128, 9 * 128, 12 * 128, 18 * 128, \
+        27 * 128, 36 * 128, 54 * 128, 108 * 128                               \
+  }
+#define SplitKKs \
+  { 65536 }
+
+#define SplitKMatmulA100Shapes ArgsProduct({SplitKMs, SplitKNs, SplitKKs})
```

Comment:
What will happen if you run these benchmarks on a non-A100 device?

Reply:
The benchmarks should run, but they won't necessarily form a single full warp. This is tricky to adapt to the host GPU since definition of tests happens at compile time via macros. I could generate all tests from say 1 to 150 blocks and then skip tests at runtime that don't divide the number of SMs evenly, but that might pollute the output since we'd be skipping a ton of tests. This is a good example where python benchmarks will be much more convenient than c++, since in that case we can generate all factors of num SMs when parametrizing the tests.

Reply:
What I wrote here is wrong. We can determine tests at runtime. I'm reworking this all to make the benchmark code a bit easier to follow now that there are matmul, partitionedk, eager, and reduction benchmarks all present there.